tests/test_ball_detector.py

```diff
@@ -0,0 +1,166 @@
+from sys import modules
+from unittest.mock import Mock
+
+
+modules_to_patch = [
+    "luma.core.interface.serial",
+    "luma.oled.device",
+    "pyinotify",
+    "pitop.camera",
+    "simple_pid",
+    "pitopcommon.smbus_device",
+    "pitopcommon.logger",
+    "pitopcommon.singleton",
+    "pitopcommon.common_ids",
+    "pitopcommon.current_session_info",
+    "pitopcommon.ptdm",
+    "pitopcommon.firmware_device",
+    "pitopcommon.command_runner",
+    "pitopcommon.common_names"
+
+]
```

Comment:
I don't think we need all of these for the ball detector. Let's remove the ones we don't need.

Reply:
@m-roberts can you take a stab at this since you know the architecture better? I'd just be blindly removing and pushing to test which might take a while